test/e2e/autoscale_test.go

```diff
@@ -192,49 +128,57 @@ func TestAutoscaleUpDownUp(t *testing.T) {
 		"CheckingEndpointAfterUpdating",
 		test.ServingFlags.ResolvableDomain)
 	if err != nil {
-		t.Fatalf(`The endpoint for Route %s at domain %s didn't serve
-			 the expected text \"%v\": %v`,
+		t.Fatalf("The endpoint for Route %s at domain %s didn't serve the expected text \"%v\": %v",
 			names.Route, domain, autoscaleExpectedOutput, err)
 	}
 
-	logger.Infof(`The autoscaler spins up additional replicas when traffic
-		    increases.`)
-	err = generateTraffic(clients, logger, 20, 20*time.Second, domain)
+	return &testContext{
+		t:              t,
+		clients:        clients,
+		logger:         logger,
+		names:          names,
+		deploymentName: deploymentName,
+		domain:         domain,
+	}
+}
+
+func assertScaleUp(ctx *testContext) {
+	ctx.logger.Infof("The autoscaler spins up additional replicas when traffic increases.")
+	err := generateTraffic(ctx, 20, 20*time.Second)
 	if err != nil {
-		logger.Fatalf("Error during initial scale up: %v", err)
+		ctx.logger.Fatalf("Error during initial scale up: %v", err)
```

Comment:
Should this be `ctx.t.Fatalf`? Otherwise this won't show up as a failed test, will it? Applies to the next occurence as  well (line 159)

Reply:
Yes, that should be t.Fatal.